lib/topology_lib_ip/library.py

```diff
@@ -24,6 +24,130 @@
 
 from ipaddress import ip_address, ip_network, ip_interface
 
+import re
+
+def parse_ip_addr_show(raw_result):
+    """
+    Parse the 'ip addr list dev' command raw output.
+
+    :param str raw_result: os raw result string.
+    :rtype: dict
+    :return: The parsed result of the show interface command in a \
+        dictionary of the form:
+
+     ::
+
+        {
+            'os_index' : '0',
+            'dev' : 'eth0',
+            'falgs_str': 'BROADCAST,MULTICAST,UP,LOWER_UP',
+            'mtu': 1500,
+            'state': 'down',
+            'link_type' 'ether',
+            'mac_address': '00:50:56:01:2e:f6',
+            'inet': '20.1.1.2',
+            'inet_mask': '24',
+            'inet6': 'fe80::42:acff:fe11:2',
+            'inte6_mask': '64'
+        }
+    """
+    # try with both inet and inet6
+    show_re = (
+        r'\s*(?P<os_index>\w+):\s+(?P<dev>\w+):\s+<(?P<falgs_str>.*)?>\s+mtu\s+(?P<mtu>\d+).+?state\s+(?P<state>\w+).+?\n'
```

Comment:
meant to say, flags_str will certianly have at least commas. I would not rule out other punctuation.